### PR TITLE
Adding functionality for a Splashscreen to be used on startup

### PR DESCRIPTION
**Pull request by james-berkheimer**
## Checklist

- [x] I have read the [CONTRIBUTING.md](../CONTRIBUTING.md) document
- [x] I formatted my changes with [black](https://github.com/psf/black)
- [x] I linted my changes with [flake8](https://github.com/PyCQA/flake8)
- [x] I have added documentation regarding my changes where necessary
- [ ] Any pre-existing tests continue to pass
- [ ] Additional tests were made covering my changes

## Types of Changes

- [ ] Bugfix (change that fixes an issue)
- [x] New Feature (change that adds functionality)
- [x] Documentation Update (if none of the other choices apply)

## Proposed Changes

Adding functionality that supports usage of a splash screen for long load times.

### CODE REVIEW

File: hab_gui/cli.py [hunk 1]
```diff
@@ -3,31 +3,42 @@
...
     if not app:
         # Otherwise create a new QApplication instance
         app = QApplication([])
+        if splash:
+            try:
+                splash_image = utils.get_splash_image(settings.resolver)
+                _splash = SplashScreen(splash_image)
+                _splash.show()
+            except Exception:
+                print("No splash screen configured")
```
Comment (MHendricks) [inline hab_gui/cli.py:39]: This is too generic, there should be a fairly specific exception being raised if no splashscreens are defined. Also having a print here essentially makes configuring splashscreens required so you don't always get that message being printed.

File: hab_gui/utils.py [hunk 2]
```diff
@@ -59,3 +62,28 @@ def make_button_coords(button_list, wrap_length, arrangement):
...
+    to any image or image directory.  Those will be distilled down to a list of
+    valid images that can be used by the Hab-Gui SplashScreen class.  This method
+    will then randomly choose an image from that list.
+    """
+    resolved_list = []
+    valid_extentions = [".jpg", ".png", ".gif"]
+    for item in resolver.site.get("splash_screen", {}):
+        if os.path.isdir(item):
+            for scan_obj in os.scandir(item):
+                if scan_obj.is_file:
```
[the remark this answers is not in the document]
Comment (james-berkheimer) [reply hab_gui/utils.py:83]: Yeah I missed this.  But oddly the function still worked.

File: hab_gui/utils.py [hunk 3]
```diff
@@ -59,3 +62,28 @@ def make_button_coords(button_list, wrap_length, arrangement):
...
+
+    Valid image formats: JPG, PNG, GIF
+
+    A resolver object will be used to grab a site.json config that defines paths
+    to any image or image directory.  Those will be distilled down to a list of
+    valid images that can be used by the Hab-Gui SplashScreen class.  This method
+    will then randomly choose an image from that list.
+    """
+    resolved_list = []
+    valid_extentions = [".jpg", ".png", ".gif"]
```
[the remark this answers is not in the document]
Comment (james-berkheimer) [reply hab_gui/utils.py:79]: I knew this was kind of half-assed and you would have a better method. :)

**Comment by MHendricks** [comment]
Also, @james-berkheimer, looks like you have to re-sign the CLA now that you have switched git accounts

**Review by MHendricks — APPROVED** [review]

**Pull request merged**